lib/Conversion/ImportVerilog/Structure.cpp

```diff
@@ -35,6 +35,40 @@ convertProcedureKind(slang::ast::ProceduralBlockKind kind) {
   llvm_unreachable("all procedure kinds handled");
 }
 
+static moore::NetKind convertNetKind(slang::ast::NetType::NetKind kind) {
+  switch (kind) {
+  case slang::ast::NetType::Supply0:
+    return moore::NetKind::Supply0;
+  case slang::ast::NetType::Supply1:
+    return moore::NetKind::Supply1;
+  case slang::ast::NetType::Tri:
+    return moore::NetKind::Tri;
+  case slang::ast::NetType::TriAnd:
+    return moore::NetKind::TriAnd;
+  case slang::ast::NetType::TriOr:
+    return moore::NetKind::TriOr;
+  case slang::ast::NetType::TriReg:
+    return moore::NetKind::TriReg;
+  case slang::ast::NetType::Tri0:
+    return moore::NetKind::Tri0;
+  case slang::ast::NetType::Tri1:
+    return moore::NetKind::Tri1;
+  case slang::ast::NetType::UWire:
+    return moore::NetKind::UWire;
+  case slang::ast::NetType::Wire:
+    return moore::NetKind::Wire;
+  case slang::ast::NetType::WAnd:
+    return moore::NetKind::WAnd;
+  case slang::ast::NetType::WOr:
+    return moore::NetKind::WOr;
+  case slang::ast::NetType::Interconnect:
+  case slang::ast::NetType::Unknown:
+  case slang::ast::NetType::UserDefined:
+    break;
+  }
+  llvm_unreachable("unsupported net kind");
```

Comment:
`llvm_unreachable` is weird in that it doesn't really print an error message in non-debug/non-assertion builds. It will just hint the compiler at this branch being unreachable in release builds. However, you do want to report an error to the user here. Something like the following:
```
static FailureOr<moore::NetKind> convertNetKind(slang::ast::NetType::NetKind kind, Location loc) {
  switch (kind) {
  // ...
  case slang::ast::NetType::Interconnect:
  case slang::ast::NetType::Unknown:
  case slang::ast::NetType::UserDefined:
    return mlir::emitError(loc, "unsupported net type `") << slang::ast::toString(kind) << "`";
  }
}
```